### PR TITLE
Node.js vs Python for backend Development

### LINKED ISSUE #1907: Node.js vs Python: Which is the best Technology for backend Development?
### Proposed title of article
Node.js vs Python: Which is the best Technology for backend Development?

### Introduction paragraph (2-3 paragraphs):
Node.js and Python are both amazing backend technologies for building dynamic web apps. The two of them fall under similar classification, yet their features are very different. There are those who argue that Node.js is better due to its high speed of execution while others choose Python because of its learning curve being shallow.

Before we argue that one is better than the other, let's first look at the primary differences between the two.

Node.js is an open-source, cross-stage, JavaScript runtime environment, based on Chrome's V8 JavaScript engine. The purpose of this engine is to compile JavaScript functions into machine code. It is well-known for its high speed and constantly advancing performance.

Python is a general-purpose programming language. It is used in Web Development to create dynamic websites using frameworks like Flask, Django, and Pyramid.

In this article, we are going to compare Node.js and Python for backend programming.

### Key takeaways:
In this article, we are going to look at the differences between Node.js and Python  based on:

1. Speed - which has the highest executing speed.
2. Universality - different ways in which the two languages can be used
3. Learning curve - which language is easier to learn and implement.
4. Extensibility - which one is easier to extend using plugins and packages
5. Syntax - which one offers developer friendly syntax

### References:
N/A

### PR DESCRIPTION
**Pull request by Erastuskanyatta**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
This fixes #1907

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
LOL we assigned ourselves at the same time @collins-a

**Comment by collins-a** [comment]
@ahmadmardeni1 no worries sir. Happens ever so often

File: articles/node.js-vs-python/index.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+Unlike Python, Node.js uses different modules for its functionality. Therefore, for the increase of scalability of an app developed by Node.js few modules are added. This will increase the demand for using Node.js by developers.
+
+Node.js is an asynchronous technology. (can run multiple processes)Hence it facilitates multithreading.
+#### Python
+Python does not support multithreading. Therefore, scalability is not easy. For Python to have easy scalability, libraries have to be used. However, this does not mean that it can compete with Node.js in terms of scalability.
+
+### UNIVERSALITY
+
+ #### Node.Js
+Node.js is used in backend development. For frontend development, developers prefer JavaScript. This saves on resources since only one team with Node.js skills is hired. This feature of Node.js has, in turn, led many developers to choose it.
```
Comment (collins-a) [inline articles/node.js-vs-python/index.md:70]: Please expound on this point on saving resources and hiring one team as it's not that clear.

File: articles/node.js-vs-python/index.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+Node.js is cross-stage. This means that a developer can create one desktop app that will work On different operating systems like MAC OS, Linux, and Windows.  This will be a great advantage to the developer in terms of the project cost.
+
+#### Python
+Python is a full-stack language. It is used in backend development while its frameworks are used in frontend development. 
+
+A python program can be written in MAC OS and the same program run in Linux, therefore Python is also a cross-stage
+
+Python is a good language for web development as well as desktop development, but unlike Node.js it is never used in mobile app development.
```
Comment (collins-a) [inline articles/node.js-vs-python/index.md:79]: I can argue out that, although to a much lesser extent, Python can be, and is used for mobile app development. Please restructure this statement to remove the word 'never'.

File: articles/node.js-vs-python/index.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+For instance, Python can be incorporated with code editors like Sublime Text, which offers some additional editing features and punctuation extensions.
+
+The great extensibility of Python and the use of many frameworks have made Python be such a great backend language that every developer would desire to use. Python frameworks include:
+
+1. [Django](https://docs.djangoproject.com/en/3.2/)
+
+2. [Flask](https://flask.palletsprojects.com/en/1.1.x/)
+
+3. [Web2Py](https://loopback.io/doc/)
```
Comment (collins-a) [inline articles/node.js-vs-python/index.md:100]: This link leads to Loopback.js as opposed to Web2Py

File: articles/node.js-vs-python/index.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+### Node.Js
+Node.js enables asynchronous. A process can be called even if another event is occurring. This means that no process is ever blocked. The different processes can be processed.
+
+Node.js is event-driven(can respond to the user in various ways). Therefore it is suitable for the development of games.
+
+### Python
+Python is not event-driven. To build an event-driven app using Python, you need to install a tool like [CPython](https://stackoverflow.com/questions/17130975/python-vs-cpython).
+
+Python enables synchronous. This makes it slow and therefore it is not suitable for projects that require a high speed of execution. So if you want to create a game program you'd rather not choose Python but Node.js
+
```
Comment (collins-a) [inline articles/node.js-vs-python/index.md]: Correct me if I'm wrong but, although less common, can't Python also carry out an asynchronous paradigm? Rephrase this statement to reflect such.

File: articles/node.js-vs-python/index.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+Node.js is event-driven(can respond to the user in various ways). Therefore it is suitable for the development of games.
+
+### Python
+Python is not event-driven. To build an event-driven app using Python, you need to install a tool like [CPython](https://stackoverflow.com/questions/17130975/python-vs-cpython).
+
+Python enables synchronous programming. This makes it slow and therefore it is not suitable for projects that require a high speed of execution. So if you want to create a game program you'd rather not choose Python but Node.js
+
+### Wrapping up
+From the above comparison,  it is evident that one software may work on one project but may not work on another different project. Thus, when choosing between Node.js and  Python, it is important to consider the needs and requirements of the two languages. I now believe that it will be easy to choose either of the two languages when working on the backend side. However, this doesn't imply that Python and Node.js are the most ideal
```
Comment (collins-a) [inline articles/node.js-vs-python/index.md:119]: The last sentence seems unfinished. Is it a full stop that's missing? Or was a thought cut off?

Also, might be of greater value to add a reference or two, or links to assist readers to read further on similar concepts to the ones you have covered.

**Review by collins-a — CHANGES_REQUESTED** [review]
@Erastuskanyatta good article. Handle the handful of changes requested and we're good to go. Cheers

**Comment by collins-a** [comment]
@Erastuskanyatta also ensure that your hero image is as close to 600*400 as possible. Hoping it is copyright free as well

**Comment by Erastuskanyatta** [comment]
@collins-a  changes completed.  Please check out my last commit.

**Review by collins-a — APPROVED** [review]
@Erastuskanyatta changes approved. 

But do take note of the commits you make as you make changes. The force-push you did made the commits I made disappear. For example, I had to re-do processes such as adding front matter. 

Until next time, good luck.

**Comment by hectorkambow** [comment]
plagiarism check done

**Pull request merged**